compiler/luci/pass/src/SubstituteTransposeToReshapePass.test.cpp

```diff
@@ -105,8 +91,15 @@ TEST_F(SubstituteTransposeToReshapeTest, simple_case)
   ASSERT_EQ(201, new_shape->at<loco::DataType::S32>(3));
 }
 
-TEST_F(SubstituteTransposeToReshapeTest_NEG, simple_case_NEG)
+TEST_F(SubstituteTransposeToReshapeTest, simple_case_NEG)
 {
+  // Create graph that tranpose input {126, 201, 1, 1} with permutation {2, 1, 3, 0}
+  buildGraph({126, 201, 1, 1}, std::vector<int32_t>({2, 1, 3, 0}));
+  // on this input shape and permutation value, output shape will be {1, 201, 1, 126}
+  // the value in buffer also change as index become reversed
+  // for example, index 1 value in input will value.at[0][1][0][0], output will be
+  // value.at[0][0][0][1] which is same with input value.at[1][0][0][0]. so this transpose cannot
+  // convert to reshape.
```

Comment:
it's a bit difficult to understand `value.at[0][1][0][0]` and `value.at[0][0][0][1]` ... maybe add some comments about `[0][1][0][0]` and `[0][0][0][1]` ?

> index 1 value in input will value.at[0][1][0][0]`

Something hard to understand... is this something like `input.at[0][1][0][0]` ?

> which is same with input value.at[1][0][0][0]

how does this work?

Reply:
Maybe original intention of the comment is following...?
```
  // With this input shape and permutation values, output shape will be {1, 201, 1, 126}.
  // The sequence of values in buffer is also changed as index becomes reversed.
  // For example, output[1][2][3][4] is not input[1][2][3][4] but input[3][2][4][1].
  // So this Transpose op cannot be converted to Reshape op.
```

Reply:
> it's a bit difficult to understand `value.at[0][1][0][0]` and `value.at[0][0][0][1]` ... maybe add some comments about `[0][1][0][0]` and `[0][0][0][1]` ?
> 

I want to mean a value in  input tensor or output tensor with that position. 

> Something hard to understand... is this something like `input.at[0][1][0][0]` ?

I think those means only input value. but I think this make more confused. so, I will update comments with `input.at` and `output.at`

> how does this work?

On transpose operation, it will change a value shape with permutation value. on this comment, it will only target on this example, so it change shape {126, 201, 1, 1} to {1, 201, 1, 126}. so output.at[0][0][0][1] will be same with input.at[1][0][0][0].

Reply:
I mean, there are two '1' in input shape and output shape, and you used index `[1]` 
If you can you can use something like `{32, 64, 16, 1}` then what shape comes from the output?

Order of `{126, 201, 1, 1} to {1, 201, 1, 126}` is `input - output` and ordef of `output.at[0][0][0][1] will be same with input.at[1][0][0][0]` is `output - input` : this made me a bit confusing. it would be better to explain in same order.

Reply:
IMHO, the sentence after `for example,` makes things confusing. Actually this test is already an example of the algorithm, so the example in the comment seems redundant.

How about removing the example part as below ? (I slightly modified @llFreetimell  's suggestion)

``` cpp
  // With this input shape and permutation values, output shape will be [1, 201, 1, 126].
  // The order of non-one values is changed (126, 201) -> (201, 126).
  // So this Transpose op cannot be converted to Reshape op.
```

Reply:
> IMHO, the sentence after `for example,` makes things confusing. Actually this test is already an example of the algorithm, so the example in the comment seems redundant.
> 
> How about removing the example part as below ? (I slightly modified @llFreetimell 's suggestion)
> 
> ```c++
>   // With this input shape and permutation values, output shape will be [1, 201, 1, 126].
>   // The order of non-one values is changed (126, 201) -> (201, 126).
>   // So this Transpose op cannot be converted to Reshape op.
> ```

I think this comment is much more clear and explain what I want to say.